c10/cuda/CUDACachingAllocator.cpp

```diff
@@ -193,11 +197,25 @@ struct THCCachingAllocator
     Block* block = NULL;
     Block* remaining = NULL;
 
-    auto it = free_blocks.lower_bound(&search_key);
-    if (it != free_blocks.end() && (*it)->device == device && (*it)->stream == stream) {
-      block = *it;
-      free_blocks.erase(it);
-    } else {
+    for (int i = 0; i < 2; ++i) {
+      if (block == NULL) {
```

Comment:
If you're going to `break` whenever you set `block = *it`, you don't need to test if `block == NULL` here right?